component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+ */
+@Extension(
+        name = "fileeventlistener",
+        namespace = "source",
+        description = "" +
+                "Fileeventlistener provides the functionality for user to get the details of files " +
+                "which have been created or modified or deleted in the execution time" ,
+
+        parameters = {
+                @Parameter(
+                        name = "uri",
+                        description =
+                                "Used to specify a file to be processed. \nThe 'uri'  should be provided.\n" +
+                                        "This uri MUST have the respective protocol specified.\n",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "monitoring.interval",
+                        description = "This parameter is used to specify the time interval (in milliseconds) " +
+                                " that the process monitor the changes for.\n",
+                        type = {DataType.STRING},
+                        optional = true,
+                        defaultValue = "1000"
+                ),
+        },
+        examples = {
+                @Example(
+                        syntax = "" +
+                                "@source(type='fileeventlistener', uri='file://abc/xyz.txt',@map(type='json')) \n" +
+                                "define stream CreateFileStream (filepath string, length long, " +
+                                "last_modified string, status string);" +
+                                "@sink(type='log')" +
+                                "define stream FooStream (filepath string, length int, " +
+                                "last_modified string, status string); \n",
+
+                        description = "" +
+                                "Under above configuration, After monitoring has been started, If the specific " +
+                                "file given in the uri gets modified or deleted. an event is created" +
+                                " with the filepath, length, last modified date and status of the file. " +
+                                "Then that will be received by the FooStream..\n"
+                ),
+                @Example(
+                        syntax = "" +
+                                "@source(type='fileeventlistener',uri='file://abc/xyz',@map(type='json')) \n" +
+                                "define stream CreateFileStream (filepath string, length long, " +
+                                "last_modified string, status string);\n" +
+                                "@sink(type='log')" +
+                                "define stream FooStream (filepath string, length int, " +
+                                "last_modified string, status string); \n",
+
+                        description = "" +
+                                "Under above configuration, all the files in the specific uri " +
```

Comment:
""Under above configuration, An events is triggered if any file under the given URI gets created, modified or deleted."

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/3f9a2a0eb50655821dd322688aa034be33ef8860